### PR TITLE
Convert Assert<String> extensions into Assert<CharSequence> extensions.

### PR DESCRIPTION
**Pull request by Kritarie**
Most of the diff is a simple cut-paste from `strings.kt` and `StringTest.kt` to `charsequence.kt` and `CharSequence.kt`.
Slight modifications were required for `isEqualTo` and `isNotEqualTo`, as they both relied on the `String?` extension of `.equals` in the standard lib.
Also adds two tests that caught a bug I introduced with this change, specifically when comparing `null` to `"null"`

Couple thoughts:
1. Is there anything I'm missing that might cause a maintenance headache with this change?
2. Are there cases where this will break compilation for consumers?

### CODE REVIEW

File: assertk/src/commonMain/kotlin/assertk/assertions/charsequence.kt [hunk 1]
```diff
@@ -53,3 +54,122 @@ fun Assert<CharSequence>.hasSameLengthAs(other: CharSequence) = given { actual -
...
+
+/**
+ * Asserts the char sequence is equal to the expected char sequence.
+ * @param ignoreCase true to compare ignoring case, the default if false.
+ * @see [isNotEqualTo]
+ */
+fun Assert<CharSequence?>.isEqualTo(other: CharSequence?, ignoreCase: Boolean = false) = given { actual ->
+    val actualString = actual?.toString()
+    val otherString = other?.toString()
+    if (actualString.equals(otherString, ignoreCase)) return
```
Comment (evant) [inline assertk/src/commonMain/kotlin/assertk/assertions/charsequence.kt:75]: So it's possible for a `CharSequence` to not be equal but their `toString()` representations to be equal (Think spanned strings on android). Think we need to keep these only for `String`.

File: assertk/src/commonMain/kotlin/assertk/assertions/charsequence.kt [hunk 2]
```diff
@@ -53,3 +54,122 @@ fun Assert<CharSequence>.hasSameLengthAs(other: CharSequence) = given { actual -
...
+    fail(other, actual)
+}
+
+/**
+ * Asserts the char sequence is not equal to the expected char sequence.
+ * @param ignoreCase true to compare ignoring case, the default if false.
+ * @see [isEqualTo]
+ */
+fun Assert<CharSequence?>.isNotEqualTo(other: CharSequence?, ignoreCase: Boolean = false) = given { actual ->
```
Comment (evant) [inline assertk/src/commonMain/kotlin/assertk/assertions/charsequence.kt:84]: Ditto for `isNotEqualTo`

File: assertk/src/commonMain/kotlin/assertk/assertions/charsequence.kt (hunk 1 above)
Comment (Kritarie) [reply assertk/src/commonMain/kotlin/assertk/assertions/charsequence.kt:75]: Good point :+1:
